app/components/avo/base_component.rb

```diff
@@ -43,4 +43,21 @@ def reflection_parent_resource
   rescue
     nil
   end
+
+  def parent_or_child_resource
+    return @resource unless link_to_child_resource_is_enabled?
+
+    ::Avo::App.get_resource_by_model_name(@resource.model.class).dup
+  end
+
+  def link_to_child_resource_is_enabled?
+    return field_linked_to_child_resource? if @parent_resource
+
+    @resource.link_to_child_resource
```

Comment:
This seems to always return `false`. Can we rewrite it somehow and remove the dependency to `@resource.link_to_child_resource`?

Reply:
it's still needed. This option allows the global linking of parent resource instead of child resource
when the resource is targeted directly without the usage of a field.

Reply:
ok. and where do we set it to true? I can't seem to find that.

or maybe I didn't really understood the use-case. can you give me an example of when we'd set it to `true`?

Reply:
I just pushed the changes in the people resource to see how to enable this feature on resource level. 
with the option toggled true.

Navigate to http://127.0.0.1:3000/admin/resources/people and edit or show the records.
It will use the child resource instead of the parent resource automatically. 
Hope it made it clear ?

Reply:
Ok, I see that now, and how it works. However, I'm pondering if this configuration switch is really worth it that much.

I'll explain what I mean.

1. I try to keep only the necessary configuration on each entity (`BaseResource`). We added and added things (and we'll add more in the future), that might make it more difficult to manage and it becomes a "God object". So efficiency is taken into account here.
1. Do you really need to toggle that switch on the resource level? Are the savings worth it? I mean, if you don't declare it on the resource you'll declare it on the field/fields. How many fields are there anyway? Maybe 10 in a truly large app? But usually just one or two places in a regular app?
1. Is this API really helpful. One developer might be used to declaring this option on the field and want to make some changes, but another developer in the team declared it on the resource. So developer 1. is trying to troubleshoot why it's not working like they are used to it because they can't seem to find it on the field. I know I hit this snag in the past.

I guess, what I'm trying to say is that I want to avoid having two ways of doing the same thing. It adds more maintenance, and the gains aren't really that high.

I'd love to hear your thoughts about this.

Reply:
Hi Adrian 

Thanks for your comment, I really appreciate the discussion and your patience.

Just to be clear:
**The field option link_to_child_resource will not override the resource option**


However, I disagree with your argumentation and I will try to answer your questions, and explain my point. 

1. I see what you mean, and I totally agree with the approach to minimize the configuration.
2. **Yes**. I don't see any other way to set the resource to link to the child resource instead of the parent resource (which is now the default behavior), when no fields declaration is involved at all.
If you have any idea how to force for example  ->//admin/resources/people to show a list of records, each linked to their respective child resources, we can do it without the resource level configuration, I am total not keen about it. 

3. Your assumption here, that the filed option will override the resource option, is not correct. 

- if developer 1 set the resource option to true, it will be used **only** when we point the browser to show page of the parent resource.
- if the second developer set the field option, we will respect the option set in the field or use the default behavior.

Reply:
Oh, ok. Your answer on point 2. really made it click for me. I always saw it like, "you only need to split the resources on the `has_many` field level, not on the "root" `/admin/resources/people`.

Thanks for explaining! That makes sense.